typescript/phoenix-sdk/examples/swap.ts

```diff
@@ -54,69 +57,16 @@ export async function swap() {
     "% slippage"
   );
 
-  const baseAccount = PublicKey.findProgramAddressSync(
-    [
-      trader.publicKey.toBuffer(),
-      TOKEN_PROGRAM_ID.toBuffer(),
-      marketData.header.baseParams.mintKey.toBuffer(),
-    ],
-    ASSOCIATED_TOKEN_PROGRAM_ID
-  )[0];
-  const quoteAccount = PublicKey.findProgramAddressSync(
-    [
-      trader.publicKey.toBuffer(),
-      TOKEN_PROGRAM_ID.toBuffer(),
-      marketData.header.quoteParams.mintKey.toBuffer(),
-    ],
-    ASSOCIATED_TOKEN_PROGRAM_ID
-  )[0];
-  const logAuthority = PublicKey.findProgramAddressSync(
-    [Buffer.from("log")],
-    Phoenix.PROGRAM_ID
-  )[0];
-
-  const orderAccounts = {
-    phoenixProgram: Phoenix.PROGRAM_ID,
-    logAuthority,
-    market: marketAddress,
-    trader: trader.publicKey,
-    baseAccount,
-    quoteAccount,
-    quoteVault: marketData.header.quoteParams.vaultKey,
-    baseVault: marketData.header.baseParams.vaultKey,
-  };
-
-  const orderPacket = Phoenix.getMarketSwapOrderPacket({
-    marketData,
+  // Generate an IOC order packet
+  const orderPacket = market.getSwapOrderPacket({
     side,
     inAmount,
     slippage,
   });
-
-  const swapIx = Phoenix.createSwapInstruction(orderAccounts, {
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore TODO why is __kind incompatible?
-    orderPacket: {
-      __kind: "ImmediateOrCancel",
-      ...orderPacket,
-    },
-  });
-
-  const controlTx = new Transaction().add(swapIx);
-
-  const swapTx = Phoenix.getMarketSwapTransaction({
-    marketAddress,
-    marketData,
-    trader: trader.publicKey,
-    side,
-    inAmount,
-    slippage,
-  });
-
-  if (JSON.stringify(controlTx) !== JSON.stringify(swapTx))
-    throw Error(
-      "Manually created transaction does not match the one created by the SDK"
-    );
+  // Genearte a swap instruction from the order packet
+  const swapIx = market.createSwapInstruction(orderPacket, trader.publicKey);
```

Comment:
Look at the code reduction!

Reply:
Not having to do accounts yourself is so nice